src/parquet/file_writer.h

```diff
@@ -39,6 +39,10 @@ class GroupNode;
 
 }  // namespace schema
 
+// RowGroupWriter implementation that optimizes memory requirement
+// All columns must be written one after the other
+// Writing to a new column prevents modification to the previous column
```

Comment:
Add `\brief` here and use `///` so these comments will show up in doxygen. We can make sure the Parquet docs show up well in http://arrow.apache.org/docs/cpp/ after the merge